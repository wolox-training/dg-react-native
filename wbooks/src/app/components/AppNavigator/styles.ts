import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

export default StyleSheet.create({
  header: {
    height: 90
  },
  headerTitle: {
    fontSize: 17,
    fontWeight: '600',
    color: COLORS.white
  }
});
