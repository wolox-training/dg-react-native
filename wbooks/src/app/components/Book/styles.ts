import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: COLORS.white,
    flexDirection: 'row',
    paddingVertical: 15,
    paddingHorizontal: 30,
    marginVertical: 5,
    borderRadius: 5,
    shadowColor: COLORS.black,
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.05,
    shadowRadius: 1,
    elevation: 1
  },
  image: {
    height: 60,
    width: 39
  },
  descriptionContainer: {
    marginLeft: 20
  },
  title: {
    fontSize: 17,
    color: COLORS.gray,
    fontWeight: '600'
  },
  author: {
    fontSize: 15,
    color: COLORS.gray
  }
});
