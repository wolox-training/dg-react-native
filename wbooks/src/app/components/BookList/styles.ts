import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: COLORS.background,
    flex: 1,
    paddingVertical: 10
  },
  subContainer: {
    flex: 1,
    paddingHorizontal: 20
  },
  emptyContainer: {
    flex: 1,
    justifyContent: 'center'
  }
});
