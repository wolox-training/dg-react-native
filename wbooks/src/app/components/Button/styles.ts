import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

const ALIGN_POSITION = 'center';

export default StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'center'
  },
  animatedStyle: {
    minWidth: 44,
    borderRadius: 22,
    marginVertical: 5
  },
  buttonDefault: {
    height: 44,
    width: '100%',
    borderRadius: 22,
    justifyContent: ALIGN_POSITION,
    alignItems: ALIGN_POSITION,
    borderWidth: 1,
    borderColor: COLORS.secondary
  },
  titleDefault: {
    fontSize: 15
  },
  titlePrimary: {
    color: COLORS.secondary
  },
  titleSecondary: {
    color: COLORS.white
  },
  iconContainer: {
    position: 'absolute'
  },
  icon: {
    tintColor: COLORS.white
  }
});
