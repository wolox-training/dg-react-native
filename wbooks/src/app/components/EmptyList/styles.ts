import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

export default StyleSheet.create({
  container: {
    alignItems: 'center'
  },
  title: {
    marginTop: 20,
    fontSize: 15,
    fontWeight: '600'
  },
  description: {
    marginTop: 12,
    fontSize: 15,
    color: COLORS.light
  }
});
