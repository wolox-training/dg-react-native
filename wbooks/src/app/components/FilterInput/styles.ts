import { StyleSheet, Platform, Dimensions } from 'react-native';
import { COLORS } from '@constants/colors';

const { width } = Dimensions.get('window');
const searchIconSize = 13;
const closeIconSize = 17;

export default StyleSheet.create({
  container: {
    // 58 is the width value of the back button
    width: width - 58,
    paddingLeft: 13
  },
  subContainer: {
    borderRadius: 14,
    paddingHorizontal: 9,
    backgroundColor: COLORS.white,
    flexDirection: 'row',
    alignItems: 'center'
  },
  searchIcon: {
    width: searchIconSize,
    height: searchIconSize
  },
  closeIcon: {
    width: closeIconSize,
    height: closeIconSize,
    tintColor: COLORS.disabled
  },
  iconEnabled: {
    tintColor: COLORS.danger
  },
  input: {
    paddingHorizontal: 13,
    flex: 1,
    height: 28,
    paddingVertical: Platform.select({
      android: 4
    })
  }
});
