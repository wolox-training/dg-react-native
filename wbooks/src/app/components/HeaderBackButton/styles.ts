import { Platform, StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    marginLeft: Platform.select({
      ios: 15
    })
  }
});
