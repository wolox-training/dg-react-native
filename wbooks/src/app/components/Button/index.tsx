import React, { useState } from 'react';
import {
  TouchableOpacity,
  TouchableOpacityProps,
  Text,
  View,
  GestureResponderEvent,
  Image
} from 'react-native';
import Animated, { withTiming, useAnimatedStyle } from 'react-native-reanimated';
import { COLORS } from '@constants/colors';
import icMyrentals from '@assets/general/ic_myrentals.png';

import styles from './styles';

type ButtonType = 'primary' | 'secondary';

interface Props extends TouchableOpacityProps {
  type?: ButtonType;
  title: string;
  activeAnimation?: boolean;
}

const DURATION = 300;
const BG_CHANGE_DURATION = 600;
const ICON_SHOW_DURATION = 900;
const INITIAL_WIDTH = 275;
const ANIMATION_WIDTH = 44;

function Button({ type, title, onPress, activeAnimation, ...props }: Props) {
  const titleStyle = type === 'secondary' ? styles.titleSecondary : styles.titlePrimary;
  const animatedBG = type === 'secondary' ? COLORS.secondary : COLORS.white;
  const [width, setWidth] = useState(INITIAL_WIDTH);
  const [color, setColor] = useState(animatedBG);
  const [titleOpacity, setTitleOpacity] = useState(1);
  const [iconOpacity, setIconOpacity] = useState(0);
  const animatedStyle = useAnimatedStyle(() => ({
    width: withTiming(width, { duration: DURATION }),
    backgroundColor: withTiming(color, { duration: BG_CHANGE_DURATION })
  }));
  const titleOpacityStyle = useAnimatedStyle(() => ({
    opacity: withTiming(titleOpacity, { duration: DURATION })
  }));
  const iconOpacityStyle = useAnimatedStyle(() => ({
    opacity: withTiming(iconOpacity, { duration: ICON_SHOW_DURATION })
  }));

  const onHandlerPress = (e: GestureResponderEvent) => {
    if (activeAnimation) {
      setWidth(ANIMATION_WIDTH);
      setColor(COLORS.success);
      setTitleOpacity(0);
      setIconOpacity(1);
    }
    if (onPress) {
      onPress(e);
    }
  };

  return (
    <View style={styles.container}>
      <Animated.View style={[styles.animatedStyle, animatedStyle]}>
        <TouchableOpacity
          accessibilityLabel="btn"
          onPress={onHandlerPress}
          {...props}
          style={[styles.buttonDefault]}>
          <Animated.View style={titleOpacityStyle}>
            <Text accessibilityLabel="btnTitle" style={[styles.titleDefault, titleStyle]}>
              {title}
            </Text>
          </Animated.View>
          <Animated.View style={[styles.iconContainer, iconOpacityStyle]}>
            <Image style={styles.icon} resizeMode="stretch" source={icMyrentals} />
          </Animated.View>
        </TouchableOpacity>
      </Animated.View>
    </View>
  );
}

export default Button;
