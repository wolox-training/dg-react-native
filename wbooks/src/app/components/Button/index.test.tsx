import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import styles from './styles';

import Button from './index';

const DEFAULT_TOUCHABLE_STYLE = { opacity: 1 };

describe('Button test suits', () => {
  const props = {
    title: 'CLICK ME',
    onPress: jest.fn()
  };

  it('Render correctly', () => {
    const button = render(<Button {...props} />);
    expect(button).toMatchSnapshot();
  });
  it('Render with correct title and primary styles', () => {
    const { getByLabelText, getByText } = render(<Button {...props} />);
    const button = getByLabelText('btn');
    const title = getByLabelText('btnTitle');
    expect(button).toBeTruthy();
    expect(button.props.style).toEqual({
      ...styles.buttonDefault,
      ...styles.buttonPrimary,
      ...DEFAULT_TOUCHABLE_STYLE
    });
    expect(title).toBeTruthy();
    expect(getByText(props.title)).toBeTruthy();
    expect(title.props.style).toEqual([styles.titleDefault, styles.titlePrimary]);
  });
  it('Render with secondary styles', () => {
    const { getByLabelText } = render(<Button type="secondary" {...props} />);
    const button = getByLabelText('btn');
    const title = getByLabelText('btnTitle');
    expect(button.props.style).toEqual({
      ...styles.buttonDefault,
      ...styles.buttonSecondary,
      ...DEFAULT_TOUCHABLE_STYLE
    });
    expect(title.props.style).toEqual([styles.titleDefault, styles.titleSecondary]);
  });
  it('Test onPress', () => {
    const { getByLabelText } = render(<Button {...props} />);
    const button = getByLabelText('btn');
    fireEvent.press(button);
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
