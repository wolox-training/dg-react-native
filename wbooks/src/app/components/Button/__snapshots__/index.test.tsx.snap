// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suits Render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <View
    animatedStyle={
      Object {
        "value": Object {
          "backgroundColor": "#FFFFFF",
          "width": "100%",
        },
      }
    }
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 22,
        "marginVertical": 5,
        "minWidth": 44,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="btn"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#00ADEE",
          "borderRadius": 22,
          "borderWidth": 1,
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 1,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          accessibilityLabel="btnTitle"
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              Object {
                "color": "#00ADEE",
              },
            ]
          }
        >
          CLICK ME
        </Text>
      </View>
      <View
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 0,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "tintColor": "#FFFFFF",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
